### DISCUSSION TITLE
Issue using Shadow Generator

### DISCUSSION THREAD

**Discussion by Jupihawx** [UMEP-dev/UMEP#659] [Q&A]
Hi everyone. I am very new so sorry if the error is obvious.
Let me explain my process. I have a point-cloud file in csv, in the format (Longitude/Latitude/Height).
I import it into QGIS, and rasterize it using TIN interpolation. So far so good, I get my DSM.

Now, when I want to use the Shadow Generator, there is an error because the estimated LON/LAT is inversed eg :

`Longitude derived from DSM: 1.3487086
Latitude derived from DSM: 103.81643
`

The thing is, when I look at the "coordinate window" in QGIS, the LAT and LON are correctly defined.
I tried to inverse the lat and lon value is the source code. It seems that it then runs with the correct sun altitude, but the generated shadow map is very strange (like a very low sun, even though the calculated altitude of the sun is close to 90°).

Is there something I am missing? 
Thanks!

**Comment by biglimp** [comment]
Make sure you have the latest version of QGIS and UMEP. If that does not solve your issue, try to change to a common CRS for your DSM such at UTM XXX. XXX depends on where on Earth you are located.

**Reply by Jupihawx** [reply]
Thanks for the answer.
Indeeed, I managed to overcome my issue after projecting my lat/lon coordinates to E/N in the correct UTM. Thanks for the input!
